src/context/index.tsx

```diff
@@ -23,10 +23,11 @@ const LDProvider: FC<Props> = ({ initialLDData, clientId, children }) => {
   // persists the data initialized server-side on the client
   const ldData = useMemo(() => initialLDData, [])
   const { user, isBot, allFlags = {} } = ldData || {}
+  const hasConsent = user && user.key !== "noConsent"
```

Comment:
I just realized this might be not the best approach will take another look tomorrow